### ISSUE TITLE
`ls` version with `Folder` and `File` arrays instead of `String`

### ISSUE THREAD

**Issue by paluh** [Thimoteus/purescript-snail#3]
I think it would be more consistent and useful if we return `{ files :: Array T.File, folder :: Array T.Folder }`. We can also provide this version as additional function like `ls'`. What do you think? Should I provide a PR?

**Comment by paluh** [comment]
EDIT: `ls` already returns full path so it should be just `map Snail.file` and `map Snail.folder` over record values ;-)

**Comment by paluh** [comment]
`ls` assumes that folder path ends with `/`. Should I treat this fact as an `Folder` type invariant?

**Comment by Thimoteus** [comment]
I like the idea of returning `File` and `Folder`. Do you mean that `ls` doesn't work if you don't end it with a trailing slash?

**Comment by paluh** [comment]
> I like the idea of returning File and Folder.

Do you want to have both versions or should I modify existing `ls` implementation?

> Do you mean that ls doesn't work if you don't end it with a trailing slash?

Yeah. Because there is just `append` in [the current implementation](https://github.com/Thimoteus/purescript-snail/blob/master/src/Snail/File.purs#L23).

**Comment by Thimoteus** [comment]
Probably best to modify the current one. Sometimes I've just added a trailing slash automatically, since I don't think I've ever had a problem with a path having more than 1 slash at the end, but I feel like dealing with paths can get pretty tricky so I never felt like putting in the effort.

Did you have an approach in mind for keeping it invariant in the type system?

**Comment by paluh** [comment]
> Did you have an approach in mind for keeping it invariant in the type system?

I don't. I'm also not sure how to provide a good balance between convenience (when I've tried `pathy` some time ago it was a bit heavy for my taste) and type safety in this case :-(

**Comment by Thimoteus** [comment]
Yeah, I thought maybe `pathy` would be good, but unfortunately it would require bridging its API to node's, which does not sound like a lot of fun ...

Maybe we can provide a function that appends a slash if there isn't one on the end, and is the identity otherwise?

**Comment by paluh** [comment]
> Maybe we can provide a function that appends a slash if there isn't one on the end, and is the identity otherwise?

Such a simple smart constructor is from my perspective the best approach :-)
